bc_obps/registration/api/user_operator.py

```diff
@@ -168,25 +169,32 @@ def is_approved_admin_user_operator(request, user_guid: str):
 @router.get("/user-operator-operator-id", response={200: UserOperatorOperatorIdOut, codes_4xx: Message})
 @authorize(["industry_user"], UserOperator.get_all_industry_user_operator_roles())
 def get_user_operator_operator_id(request):
-    user_operator = get_object_or_404(
-        UserOperator, user_id=request.current_user.user_guid, status=UserOperator.Statuses.APPROVED
-    )
+    user_operator = get_object_or_404(UserOperator, user_id=request.current_user.user_guid)
     return 200, {"operator_id": user_operator.operator_id}
 
 
+@router.get("/user-operator-id", response={200: UserOperatorIdOut, codes_4xx: Message})
+@authorize(["industry_user"], UserOperator.get_all_industry_user_operator_roles())
+def get_user_operator_id(request):
+    user_operator = get_object_or_404(UserOperator, user_id=request.current_user.user_guid)
+    print(user_operator)
+    return 200, {"user_operator_id": user_operator.id}
+
+
 @router.get(
     "/select-operator/user-operator/{int:user_operator_id}",
     response=UserOperatorOut,
 )
 @authorize(AppRole.get_all_authorized_app_roles(), UserOperator.get_all_industry_user_operator_roles())
 def get_user_operator(request, user_operator_id: int):
     user: User = request.current_user
-    user_operator = get_object_or_404(UserOperator, id=user_operator_id)
     if user.is_industry_user():
-        authorized_users = get_an_operators_approved_users(user_operator.operator)
-        if user.user_guid not in authorized_users:
-            raise HttpError(401, UNAUTHORIZED_MESSAGE)
-    return UserOperatorOut.from_orm(user_operator)
+        # Industry users can only get their own UserOperator instance
+        user_operator = get_object_or_404(UserOperator, id=user_operator_id, user=user.user_guid)
+        return UserOperatorOut.from_orm(user_operator)
+    else:
+        user_operator = get_object_or_404(UserOperator, id=user_operator_id)
+        return UserOperatorOut.from_orm(user_operator)
```

Comment:
Thanks for the great catch, now we are checking if `user.is_industry_user` and only returning a `user_operator` that they are associated with. `CAS` internal can view all user operators if they supply a valid id.